src/clsourcedependencies_parser.cc

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if NINJA_ENABLE_CL_SOURCE_DEPENDENCIES
+
+#include "clsourcedependencies_parser.h"
+
+#include "rapidjson/document.h"
+
+#include "clparser.h"
+#include "metrics.h"
+#include "util.h"
+
+bool ParseCLSourceDependencies(const StringPiece content, std::string* err,
+                               std::vector<std::string>& includes) {
+  METRIC_RECORD("CLSourceDependenciesParser::Parse");
+
+  rapidjson::Document doc;
+  if (doc.Parse(content.str_).HasParseError()) {
+    *err = std::string("sourceDependencies parse error");
```

Comment:
nit: remove the std::string() constructor calls here and all cases below for simplicity.

Reply:
Done.